fluss-client/src/main/java/org/apache/fluss/client/lookup/LookupSender.java

```diff
@@ -126,13 +126,21 @@ private void runOnce(boolean drainAll) throws Exception {
         sendLookups(lookups);
     }
 
-    private void sendLookups(List<AbstractLookupQuery<?>> lookups) {
+    private void sendLookups(List<AbstractLookupQuery<?>> lookups) throws Exception {
         if (lookups.isEmpty()) {
             return;
         }
         // group by <leader, lookup type> to lookup batches
         Map<Tuple2<Integer, LookupType>, List<AbstractLookupQuery<?>>> lookupBatches =
                 groupByLeaderAndType(lookups);
+
+        // if no lookup batches, sleep a bit to avoid busy loop. This case will happen when there is
+        // no leader for all the lookup request in queue.
+        if (lookupBatches.isEmpty()) {
```

Comment:
We should only sleep when the lookup queue is also emtpy, because currently we have a very small batch size. 

```java
if (lookupBatches.isEmpty() && !lookupQueue.hasUnDrained()) {
```